packages/sdk-metrics/src/export/PeriodicExportingMetricReader.ts

```diff
@@ -99,11 +100,24 @@ export class PeriodicExportingMetricReader extends MetricReader {
       );
     }
 
-    const result = await internal._export(this._exporter, resourceMetrics);
-    if (result.code !== ExportResultCode.SUCCESS) {
-      throw new Error(
-        `PeriodicExportingMetricReader: metrics export failed (error ${result.error})`
-      );
+    const doExport = async () => {
+      const result = await internal._export(this._exporter, resourceMetrics);
+      if (result.code !== ExportResultCode.SUCCESS) {
+        throw new Error(
+          `PeriodicExportingMetricReader: metrics export failed (error ${result.error})`
+        );
+      }
+    };
+
+    // Avoid scheduling a promise to make the behavior more predictable and easier to test
+    if (resourceMetrics.resource.asyncAttributesHaveResolved()) {
+      await doExport();
+    } else {
+      resourceMetrics.resource
+        .waitForAsyncAttributes()
+        .then(await doExport, err =>
```

Comment:
Should not `await` in place.

```suggestion
        .then(doExport, err =>
```